### DISCUSSION TITLE
Response of a thin silicon radiation sensor, Landau distribution

### DISCUSSION THREAD

**Discussion by Jaap1957** [nrc-cnrc/EGSnrc#831] [Physics] [answered]
The response of a Si based sensor looks very much like the Landau curve:
![afbeelding](https://user-images.githubusercontent.com/96918927/151163354-ada4f444-8a5b-42df-8f1e-531a52d2420e.png)

This distribution is a distribution of signal, which is proportional to the number of electron hole pairs, which again is proportional (says H. Bichsel) to deposited energy in the sensitive part (the so called EPI layer) of the sensor.
Deposited energy distribution calculated with EGS nrc looks like:
![afbeelding](https://user-images.githubusercontent.com/96918927/151164620-4aa0fe96-8900-4cf8-a764-c9cd627742d9.png)

So the MC calculation misses the smooth start (left part) of the curve.
Also the 'ringing' of the MC curve around 5 keV does not come back in the measurement..

Literature tells me there is just a simple conversion from deposited energy to # of electron hole pairs (1 eh per 3.7 eV).
Generated eh pairs are converted with some constant factor to sensor signal.
Therefore I would expect the same form of curves.

Mr Tessier already gave some thoughts to this subject in #826, which stressed the peaky form of the MC calculation even more. Thank you Mr Tessier !

The peak is due to electrons traveling past the (thin) sensitive layer without leaving a trace ... because all their interactions are below cutoffs. Is there still a model of these energy transitions so we can fill the part between 0 and 4 keV ? And get rid of the ringing?

Thanks again.

**Comment by ftessier** [comment, marked as answer]
The top curve is the response from a silicon detector, but I am not sure what it represents exactly (voltage distribution?). There is a lot going on between the energy deposition events and the production of the signal: bias, drift, recombination, gain, time resolution and so on.

The Monte Carlo curve represents the distribution of energy deposited for each incident electron, so it is closer to what would be measured if you could detect a single electron at a time in the sensor; state of the art time resolution seems to be ~100 ps, corresponding to incident beam currents in the nanoamps range. It would be interesting to understand how this signal is convoluted into the detector response. To some extent, you can investigate this convolution with EGSnrc by scoring the energy of _multiple_ histories bunched together. Or else more efficiently by just sampling the one-electron distribution to create bunches "by hand". Either way, the main sharp peak will get smoothed out.

Keep in mind that the first "ringing" after the sharp peak is due to the `ae` threshold, as discussed in #826.

**Reply by ftessier** [reply]
Here are distributions obtained when scoring the energy deposited for bunches of histories (a fixed number makes sense for a generated beam; for a radioactive source, you'd use a Poisson distribution for the number of histories in a bunch).

![out](https://user-images.githubusercontent.com/1444241/151184285-4e2bc137-2655-4905-8b89-1d9a3bb13ed6.png)

**Reply by Jaap1957** [reply]
Thanks again !
This is a nice way to look at it.
The sensor i am talking about is a camera.
Therefore indeed single histories are scored, without need for picosec time resolution.

Nevertheless, your approach delivers a curve much more alike the measured curve. So thanks for this idea.

**Reply by ftessier** [reply]
I don't think the fact that it is a camera matters, there is still a time resolution. Unless you mean that you are capturing rare events, i.e., you are operating in the single-shower limit. Even then, I would expect the signal to be smoothed out due to the electronic chain ultimately leading to the sensor signal.

**Reply by ftessier** [reply]
For the record, you can reproduce the foregoing results with the following patch to `egs_phd` from EGSnrc v2021, commit a6fc389c:

```diff
diff --git a/egs_home/egs_phd/egs_phd.cpp b/egs_home/egs_phd/egs_phd.cpp
index 2d4b92d2..44edc5a7 100644
--- a/egs_home/egs_phd/egs_phd.cpp
+++ b/egs_home/egs_phd/egs_phd.cpp
@@ -112,0 +113,6 @@ int phd_app::initScoring() {
+    // get number of showers in a bunch
+    int mybunch = 1;
+    if (!options->getInput("spectrum bunch", mybunch)) {
+        spectrum_bunch = mybunch;
+    }
+
@@ -126,0 +133,3 @@ int phd_app::initScoring() {
+    // initialize energy of bunch
+    myEnergy = 0.0;
+
@@ -164 +172,0 @@ int phd_app::simulateSingleShower() {
-    EGS_Float myEnergy = 0.0;
@@ -170,0 +179 @@ int phd_app::simulateSingleShower() {
+      if (current_case % spectrum_bunch == 0) {
@@ -172,3 +180,0 @@ int phd_app::simulateSingleShower() {
-        if (mybin == nbin) {
-            mybin--;
-        }
@@ -177,0 +184,3 @@ int phd_app::simulateSingleShower() {
+        // reset energy of bunch
+        myEnergy = 0.0;
+      }
@@ -179,0 +189 @@ int phd_app::simulateSingleShower() {
+
@@ -290,0 +301,10 @@ void phd_app::outputResponse(EGS_ScoringArray *spec) {
+
+    // get normalization
+    double norm = 0.0;
+    for (int i=0; i<nbin; i++) {
+        double y, dy;
+        spectrum->currentResult(i, y, dy);
+        norm += y*Ebin;
+    }
+    norm = 1.0/norm;    // normalize distribution to unit integral
+
@@ -297,2 +317,2 @@ void phd_app::outputResponse(EGS_ScoringArray *spec) {
-        y = y*current_case;
-        dy = dy*current_case;
+        y = y*norm;
+        dy = dy*norm;
diff --git a/egs_home/egs_phd/egs_phd.h b/egs_home/egs_phd/egs_phd.h
index 9c2d1e1c..bf7a2dd3 100644
--- a/egs_home/egs_phd/egs_phd.h
+++ b/egs_home/egs_phd/egs_phd.h
@@ -50,0 +51,2 @@ class APP_EXPORT phd_app : public EGS_AdvancedApplication {
+    int                 spectrum_bunch;        // number of histories to bunch together for scoring spectrum
+    EGS_Float           myEnergy;              // running energy of history bunch
```

**Reply by Jaap1957** [reply]
The camera is a counting camera in a TEM where the dose rate is low compared to the number of pixels.
For example 1 electron per 2500 pixels per camera frame.
With a 4000x4000 pixel camera this leads to about 60 000 events, each event is due to a single incoming electron. The signal then in such an event is due to a single history.

Your solution is to me completely out of the box - In view of what I know I would never thought of averaging histories. Chapeau ! and thank you.

**Reply by ftessier** [reply]
I see what you mean, the large number of pixels reduces the event rate _per pixel_ (per detector); interesting! If the detection is really in the single electron per event, then perhaps one has to consider the electron digitization chain to understand why the profile is smoothed out...

**Reply by Jaap1957** [reply]
![afbeelding](https://user-images.githubusercontent.com/96918927/151308476-c87de36b-44ad-441b-8511-eac18c520dbd.png)

Here a python script to get "bunches"  from the original 1-history spectrum output of egs_phd
[stat.py.txt](https://github.com/nrc-cnrc/EGSnrc/files/7948302/stat.py.txt)

Of course you have to remove the '.txt' that github forces me to add to make it a script again.
It produces the plot above.

And you are right there must be a blur in the conversion keV >> eh pairs.

**Reply by Jaap1957** [reply]
![afbeelding](https://user-images.githubusercontent.com/96918927/151325029-5c70ffac-2e01-47eb-b1aa-5546a93b7b93.png)

Here the result when i take 0.521 as cutoff; and convert the depo energy via 3.7eV/ehpair to a number which i give a random poisson spread.
It looks like your 'bunch' idea still gives a better fit to experiment 👍

**Reply by ftessier** [reply]
Nice! Thank you for the update. Indeed, I forgot that you had mentioned 3.7 eV/pair, that it was in fact so small. Thanks for the reminder and demonstration; this can fully account for the spread, since the number of pairs created for _one_ history will have a significant spread.

**Comment by ftessier** [comment]
To clarify, from the original post:

> The peak is due to electrons traveling past the (thin) sensitive layer without leaving a trace ... because all their interactions are below cutoffs. Is there still a model of these energy transitions so we can fill the part between 0 and 4 keV ? And get rid of the ringing?

The first peak, or rather the initial flat region is **not an artifcat.** It corresponds to physical reality: no electron can physically cross the layer and loose less that the corresponding CSDA energy loss, where you find the first peak (the fact that the distribution is not 0 below the peak is due to relatively rare backscatter events).

The arttifact from the 1 keV `ae` threshold is the _depletion_ in the 1 keV following the first peak maximum, to the next minimum (the first "ringing").

**Comment by Jaap1957** [comment]
For those who share my (too low) patch abilities , here are the 2 files needed for the 'bunch' option:

[egs_phd.cpp.txt](https://github.com/nrc-cnrc/EGSnrc/files/7952617/egs_phd.cpp.txt)
[egs_phd.h.txt](https://github.com/nrc-cnrc/EGSnrc/files/7952619/egs_phd.h.txt)

Just put them instead of egs_phd.cpp and egs_phd.h.
You can then add an option like

spectrum bunch = 5

before 
:stop spectrum:
in the ...egsiinp file.

This will score the average of 5 histories instead of each history in the spectruim file.

**Reply by ftessier** [reply]
Thanks. _Note:_ To patch using the patch file, put it in your top-level `EGSnrc` directory, and run `git apply patch` (or put the patch inside the `egs_phd` directory and run `git apply -p3 patch`.
